frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/sc-node-list.tsx

```diff
@@ -17,11 +18,16 @@ import '../ocs-install.scss';
 
 const getRows: GetRows = ({ componentProps, customData }) => {
   const { data } = componentProps;
-  const { nodes, setNodes, filteredNodes } = customData;
+  const { nodes, setNodes, filteredNodes, isHostNameSame } = customData;
+  let filteredData: NodeKind[] = [];
 
   const nodeList = filteredNodes ?? data.map(getName);
-  const hostNames = createMapForHostNames(data);
-  const filteredData = data.filter((node) => nodeList.includes(hostNames[getName(node)]));
+  if (isHostNameSame) {
+    filteredData = data.filter((node) => nodeList.includes(getName(node)));
+  } else {
+    const hostNames = createMapForHostNames(data);
+    filteredData = data.filter((node) => nodeList.includes(hostNames[getName(node)]));
+  }
```

Comment:
When the hostname & node name is the same then the`hostNames` map can have the same value in key value pair? Why do we need if condition looking at line 23 & 24 it should have worked. 

Am I missing something?

Reply:
Else part will be executed only when the hostname and node names are not same. Check at line 25

Reply:
Right we can create the map for both cases(if they are diff or not) and remove the if the condition itself. Rather than passing a https://github.com/openshift/console/pull/6726/files#diff-99347a73d43d6e64cba499ad4c8093edR147 hardcoded true value.

Reply:
For i3, `filteredNodes` from `create storage cluster step` will have `hostNames` as this is fetched from pv's hostname label. Hence `filteredData` will be empty as this condition will not work - `nodeList.includes(getName(node))` as its comparing with node name. That's why the else block is required. Hope that clarifies.
Else you are welcome to debug if you want to understand further.

Reply:
Why dont you just maintain hostname, discovery requires hostname only

Reply:
Just maintaining hostname will not work for all other lists and will increase the complexity. Hence, I prefer not to.
Anyways this comment seems like more of refactoring change, which I prefer not to do as a part of this PR.